index.js

```diff
@@ -228,14 +253,32 @@ async function validateCredentials(expectedAccessKeyId) {
   }
 }
 
-function getStsClient(region) {
+function getStsClient(region, credentials) {
```

Comment:
I'm unsure what problem this is exactly trying to solve. Why do we need to pass in credentials now?

There are a couple other instances where `getStsClient` and `exportAccountId` get called (lines 43 and 382 in your index.js) without being updated to have credentials passed in. Since I'm confused at the necessity for this in the first place, I'm not sure why credentials are only passed on line 418. Could you clear this up for me?